client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx

```diff
@@ -98,12 +98,26 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goNext, goToStep, flow }: S
 	const launchTask = enhancedTasks?.find( ( task ) => task.isLaunchTask === true );
 
 	const showLaunchTitle = launchTask && ! launchTask.disabled;
-	const domainUpgradeBadgeUrl = ! site?.plan?.is_free
-		? `/domains/manage/${ siteSlug }`
-		: `/domains/add/${ siteSlug }?domainAndPlanPackage=true`;
-	const showDomainUpgradeBadge =
-		sidebarDomain?.isWPCOMDomain &&
-		! enhancedTasks?.find( ( task ) => task.id === 'domain_upsell' );
+
+	function getDomainUpgradeBadgeUrl() {
+		if ( siteIntentOption === 'start-writing' || siteIntentOption === 'design-first' ) {
```

Comment:
```suggestion
		if ( isBlogOnboardingFlow( siteIntentOption ) ) {
```

Let's reuse `isBlogOnboardingFlow` and for typescript warnings, we could do this:

```
const siteIntentOption = site?.options?.site_intent ?? null;
```

What this does is when it's `undefined`, it will return null as the fallback value.

Reply:
Thanks for your suggestions! I commited